src/main/java/frc/robot/hardware/phoenix6/Phoenix6Device.java

```diff
@@ -31,17 +31,30 @@ public boolean isConnected() {
 		return connectedInput.connected;
 	}
 
+	private InputSignal<?>[] getValidSignals(InputSignal<?>... signals) {
+		LinkedList<InputSignal<?>> signalsSet = new LinkedList<>();
+		for (InputSignal<?> signal : signals) {
+			if (signal instanceof Phoenix6SignalBuilder.SignalGetter) {
+				signalsSet.add(signal);
+			} else {
+				new Alert(
+					Alert.AlertType.WARNING,
+					logPath + "signal named " + signal.getName() + " got invalid type " + signal.getClass().getSimpleName()
+				).report();
+			}
+		}
+		return signalsSet.toArray(new InputSignal<?>[0]);
+	}
+
+
 	private StatusCode refreshSignals(InputSignal<?>... signals) {
 		LinkedList<StatusSignal<Double>> signalsSet = new LinkedList<>();
-		for (int i = 0; i < signals.length; i++) {
-			if (signals[i] instanceof Phoenix6SignalBuilder.SignalGetter signalGetter) {
+		for (InputSignal<?> signal : signals) {
+			if (signal instanceof Phoenix6SignalBuilder.SignalGetter signalGetter) {
 				signalsSet.add(signalGetter.getSignal());
-				if (signals[i] instanceof Phoenix6BothLatencySignal bothLatencySignal) {
+				if (signal instanceof Phoenix6BothLatencySignal bothLatencySignal) {
```

Comment:
does this **need** to be nested inside the other `if`?

Reply:
yes